### PR TITLE
Reworking Nimbus Experiments to compose separate PR

### LINKED ISSUE #27819: Nimbus Experiments rework to compose
Rework Nimbus experiments to be using compose.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21979)

### PR DESCRIPTION
**Pull request by vrrashkov**
image

Following [this PR](https://github.com/mozilla-mobile/fenix/pull/27660#pullrequestreview-1165864158) this is new PR only for Experiments.

1. Created a separate file for the **NimbusExperiments** composable
2. Improved the kDoc 
3. Using **TextListItem** 
4. Changed style based on **ktlint**

About the NavHost I can see that currently the project is not using the Compose NavHost so is this the idea to use that in the future or I have just missed it. 

Fixes #27819
Fixes #27819

### CODE REVIEW

**Comment by gabrielluong** [comment]
@vrrashkov Regarding NavHost, we haven't been able to use it because we were still on a version of Compose and Navigation that didn't support NavHost. The idea is to try to use this with the Nimbus Branches.

File: app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+package org.mozilla.fenix.nimbus.view
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt:1]: We're missing the license file here

File: app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * @param experiment current experiment
+ * @param onSelectedExperiment Callback when experiment is selected it returns [AvailableExperiment]
+ */
+@Composable
+fun NimbusExperimentItem(
+    experiment: AvailableExperiment,
+    onSelectedExperiment: (AvailableExperiment) -> Unit,
+) {
+    TextListItem(
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt:46]: Let's inline this - no need to have a wrapper function for `TextListItem`.

File: app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import org.mozilla.experiments.nimbus.AvailableExperiment
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * List of Nimbus Experiments,
+ *
+ * @param availableExperiments list of available experiments that are going to be displayed
+ * @param onSelectedExperiment Callback when experiment is selected it returns [AvailableExperiment]
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt:18]:
```suggestion
 * List of Nimbus Experiments.
 *
 * @param experiments List of [AvailableExperiment] that are going to be displayed.
 * @param onExperimentClick Invoked when the user clicks on an [AvailableExperiment].
```

File: app/src/main/java/org/mozilla/fenix/nimbus/NimbusExperimentsFragment.kt [hunk 4]
```diff
@@ -5,81 +5,51 @@
...
 
-                lifecycleScope.launch(Main) {
-                    runIfFragmentIsAttached {
-                        if (!shouldRefresh) {
-                            adapter = NimbusExperimentAdapter(
-                                experimentsView,
-                                experiments,
+                NimbusExperiments(
+                    availableExperiments = experiments,
+                    onSelectedExperiment = { selectedExperiment ->
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/NimbusExperimentsFragment.kt:43]:
```suggestion
                    onSelectedExperiment = { experiment ->
```

**Comment by gabrielluong** [comment]
Could we also link this PR to any of the opened issues? Otherwise, we should open an new issue to describe the work that is being completed.

**Comment by vrrashkov** [comment]
> Could we also link this PR to any of the opened issues? Otherwise, we should open an new issue to describe the work that is being completed.

There was only 2 design issues connected to Branches but because branches was connected to Experiments, I decided to do both, that was the initial idea at least. So for Experiments there was actually no issue I think.

File: app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * List of Nimbus Experiments.
+ *
+ * @param experiments List of [AvailableExperiment] that are going to be displayed.
+ * @param onExperimentClick Invoked when the user clicks on an [AvailableExperiment].
+ */
+@Composable
+fun NimbusExperiments(
+    availableExperiments: List<AvailableExperiment> = listOf(),
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/nimbus/view/NimbusExperiments.kt:26]: My prior review changes to the `@param experiments` meant to imply that I also wanted the actual parameter name to be refactored.

```suggestion
    experiments: List<AvailableExperiment> = listOf(),
```

**Comment by gabrielluong** [comment]
@vrrashkov Apologizes for the delay, I think the changes are looking good.

**Comment by gabrielluong** [comment]
> > Could we also link this PR to any of the opened issues? Otherwise, we should open an new issue to describe the work that is being completed.
> 
> There was only 2 design issues connected to Branches but because branches was connected to Experiments, I decided to do both, that was the initial idea at least. So for Experiments there was actually no issue I think.

If you don't mind, could we file a new issue and update your commit description to follow "For #XXXX - Short description".

**Comment by vrrashkov** [comment]
> > > Could we also link this PR to any of the opened issues? Otherwise, we should open an new issue to describe the work that is being completed.
> > 
> > 
> > There was only 2 design issues connected to Branches but because branches was connected to Experiments, I decided to do both, that was the initial idea at least. So for Experiments there was actually no issue I think.
> 
> If you don't mind, could we file a new issue and update your commit description to follow "For #XXXX - Short description".

Sorry for the delay, created an issue and updated the commit message.

**Comment by gabrielluong** [comment]
@vrrashkov Just for visibility, I am still looking for this requested change https://github.com/mozilla-mobile/fenix/pull/27683#discussion_r1024697404. Thanks, I will push towards landing this week.

**Review by gabrielluong — APPROVED** [review]
Will land this over in https://github.com/mozilla-mobile/fenix/pull/27683

**Pull request closed without merging**